ext/opcache/Optimizer/dce.c

```diff
@@ -106,6 +106,7 @@ static inline zend_bool may_have_side_effects(
 		case ZEND_IS_SMALLER:
 		case ZEND_IS_SMALLER_OR_EQUAL:
 		case ZEND_CASE:
+			/* TODO: Can ZEND_CAST throw when converting object to array (in custom handlers for zend_get_properties_for?) */
```

Comment:
I'd say yes. Throwing is not handled by this function though, but by zend_may_throw.

Reply:
Removed comment